### PR TITLE
+str #17702: Add Keep.none

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

**Comment by ktoso** [comment]
LGTM, I like it

**Comment by drewhk** [comment]
PLS BUILD

**Comment by drewhk** [comment]
PLS BUILD

File: akka-stream/src/main/scala/akka/stream/javadsl/Materialization.scala [hunk 1]
```diff
@@ -7,12 +7,16 @@ import akka.japi.function
...
 object Keep {
   private val _left = new function.Function2[Any, Any, Any] with ((Any, Any) ⇒ Any) { def apply(l: Any, r: Any) = l }
   private val _right = new function.Function2[Any, Any, Any] with ((Any, Any) ⇒ Any) { def apply(l: Any, r: Any) = r }
   private val _both = new function.Function2[Any, Any, Any] with ((Any, Any) ⇒ Any) { def apply(l: Any, r: Any) = new akka.japi.Pair(l, r) }
+  private val _none = new function.Function2[Any, Any, Unit] with ((Any, Any) ⇒ Unit) { def apply(l: Any, r: Any) = () }
 
   def left[L, R]: function.Function2[L, R, L] = _left.asInstanceOf[function.Function2[L, R, L]]
   def right[L, R]: function.Function2[L, R, R] = _right.asInstanceOf[function.Function2[L, R, R]]
   def both[L, R]: function.Function2[L, R, L Pair R] = _both.asInstanceOf[function.Function2[L, R, L Pair R]]
+  def none[L, R]: function.Function2[L, R, Unit] = _both.asInstanceOf[function.Function2[L, R, Unit]]
```
Comment (m-sp) [inline akka-stream/src/main/scala/akka/stream/javadsl/Materialization.scala]: shouldn't this be `_none.asInstanceOf[function.Function2[L, R, Unit]]`?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/javadsl/Materialization.scala]: yes, thanks!

**Comment by drewhk** [comment]
Fixes #17723

**Pull request merged**